pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -220,24 +220,32 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
+const (
+	ResourceSelectorAll = "*"
+)
+
+// ResourceSelector identifies the objects to be included within a PermissionClaim either by name or namespace.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.namespaces) || has(self.names)",message="at least one field must be set"
 type ResourceSelector struct {
-	// name of an object within a claimed group/resource.
+	// names of specific resources to select.
 	// It matches the metadata.name field of the underlying object.
-	// If namespace is unset, all objects matching that name will be claimed.
+	// A value of "*" or empty means all object names are permitted.
```

Comment:
this is an anti-pattern. Don't handle empty and unset differently.

Reply:
actually I misread. But it is still strange to have an empty string. Why do we need that?

Reply:
would also expect this to be defaulted.

Reply:
I don't think we need an empty string. I like the defaulting suggestion.